src/library_pthread.js

```diff
@@ -408,6 +408,7 @@ var LibraryPThread = {
   },
 
   __emscripten_thread_cleanup: function(thread) {
+    // Called when a thread needs to be cleaned up so it can be reused.
```

Comment:
Perhaps mention that its called in exactly two places: "When a detached thread exits and when an attached thread is joined"?

Reply:
Mentioned/elaborated with commit f4058e2e8aa4fdda6aed317327d654b898371c2a.